src/main/java/de/tum/in/www1/artemis/service/SubmissionService.java

```diff
@@ -303,7 +320,35 @@ public Result copyResultFromPreviousRoundAndSave(Submission submission, Result o
         Result newResult = new Result();
         newResult.setParticipation(submission.getParticipation());
         copyFeedbackToNewResult(newResult, oldResult);
+        return copyResultContentAndAddToSubmission(submission, newResult, oldResult);
+    }
+
+    /**
+     * This method is used to create a new result, after a complaint has been accepted or denied.
+     * The new result contains the updated feedback of the result the complaint belongs to.
+     *
+     * @param submission the submission where the original result and the result after the complaintResponse belong to
+     * @param oldResult the original result, before the response
+     * @param feedbacks the new feedbacks after the response
+     * @return the newly created result
+     */
+    public Result createResultAfterComplaintResponse(Submission submission, Result oldResult, List<Feedback> feedbacks) {
+        Result newResult = new Result();
+        newResult.setParticipation(submission.getParticipation());
+        copyFeedbackToResult(newResult, feedbacks);
+        newResult = copyResultContentAndAddToSubmission(submission, newResult, oldResult);
+        return newResult;
+    }
 
+    /**
+     * Copies the content of one result to another, and adds the second result to the submission.
+     *
+     * @param submission the submission which both results belong to, the newResult comes after the oldResult in the resultlist
+     * @param newResult the result where the content is set
+     * @param oldResult the result from which the content is copied from
+     * @return the newResult
+     */
+    private Result copyResultContentAndAddToSubmission(Submission submission, Result newResult, Result oldResult){
```

Comment:
If anyone has an idea how I can fix this issue, please tell me :)

Reply:
You can ignore this warning I think.